lms/djangoapps/courseware/views/views.py

```diff
@@ -863,6 +864,126 @@ def post(self, request, course_id):
         return redirect(reverse('about_course', args=[text_type(course_key)]))
 
 
+@ensure_csrf_cookie
+@cache_if_anonymous()
+def course_page(request, course_id):
+    discovery_course = get_course_from_discovery(course_id)
+    if not discovery_course:
+        raise Http404(u"Course key not found on discovery")
+
+    [course] = discovery_course
+    course_runs = course.get('course_runs')
+    course_runs_context = []
+    course_staff = []
+    course_organization = ''
+    can_user_enroll = False
+    staff_ids = set()
+
+    for course_run in course_runs:
+        course_key = CourseKey.from_string(course_run.get('key'))
+        can_self_enroll_in_course_run = can_self_enroll_in_course(course_key)
+        can_user_enroll = can_user_enroll or can_self_enroll_in_course_run
+
+        if can_self_enroll_in_course_run:
```

Comment:
this is a pretty large if condition, can we move this to a separate function for better readability.

Reply:
fixed